src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -263,9 +263,11 @@ public function configureOptions(OptionsResolver $resolver)
                 return $choices;
             }
 
-            ChoiceType::normalizeLegacyChoices($choices, $choiceLabels);
+            if (null === $choices) {
+                $choices = array();
```

Comment:
@nicolas-grekas some sub-types are using ``null`` as default value of the option to distinguish between passing choices explicitly and using other ways of building choices (using the query builder). This change would break them (the PropelType for instance).
The Doctrine entity type is not affected because it changes the ``choices_as_values`` default and so is not triggered this line of code.

IMO, if ``null`` is given as input, you should just return it as is here (and letting the validation happening later checking this case in case the subtype does not handle it properly)